Library/Services/Implementation/NfieldSurveyInterviewSimulationService.cs

```diff
@@ -0,0 +1,171 @@
+﻿//    This file is part of Nfield.SDK.
+//
+//    Nfield.SDK is free software: you can redistribute it and/or modify
+//    it under the terms of the GNU Lesser General Public License as published by
+//    the Free Software Foundation, either version 3 of the License, or
+//    (at your option) any later version.
+//
+//    Nfield.SDK is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU Lesser General Public License for more details.
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using Newtonsoft.Json;
+using Nfield.Extensions;
+using Nfield.Infrastructure;
+using Nfield.Models;
+using Nfield.Models.NipoSoftware.Nfield.Manager.Api.Models;
+using Nfield.Utilities;
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Nfield.Services.Implementation
+{
+    /// <summary>
+    /// Implementation of <see cref="INfieldSurveyInterviewSimulationService"/>
+    /// </summary>
+    internal class NfieldSurveyInterviewSimulationService : INfieldSurveyInterviewSimulationService, INfieldConnectionClientObject
+    {
+        readonly IFileSystem _fileSystem;
+
+        public NfieldSurveyInterviewSimulationService()
+        {
+            _fileSystem = new FileSystem();
+        }
+
+        public NfieldSurveyInterviewSimulationService(IFileSystem fileSystem)
+        {
+            _fileSystem = fileSystem;
+        }
+
+        /// <summary>
+        /// <see cref="INfieldSurveyInterviewSimulationService.GetHintsAsync(string)"/>
+        /// </summary>
+        public async Task<string> GetHintsAsync(string surveyId)
```

Comment:
Maybe you can also check here that the surveyId is not null or empty (just for consistency with the other methods)

Reply:
checked